### PR TITLE
refactor(cli): move `snapshot_from_lockfile` function to `deno_npm`

### PR DESCRIPTION
**Pull request by magurotuna**
This commit moves `snapshot_from_lockfile` function to [deno_npm crate](https://github.com/denoland/deno_npm). This allows this function to be called outside Deno CLI (in particular, Deno Deploy).

Depends on:
- https://github.com/denoland/deno_npm/pull/27
- https://github.com/denoland/deno_lockfile/pull/3 (transitively)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -167,6 +167,9 @@ winres = "=0.1.12"
 winapi = "=0.3.9"
 windows-sys = { version = "0.48.0", features = ["Win32_Media"] }
 
+[patch.crates-io]
+deno_npm = { git = "https://github.com/magurotuna/deno_npm.git", branch = "snapshot_from_lockfile" }
```
Comment (dsherret) [inline Cargo.toml:171]: We can update this to 0.11.0 now

File: cli/args/mod.rs [hunk 2]
```diff
@@ -819,16 +820,26 @@ impl CliOptions {
 
     if let Some(lockfile) = self.maybe_lockfile() {
       if !lockfile.lock().overwrite {
-        return Ok(Some(
-          snapshot_from_lockfile(lockfile.clone(), api)
```
Comment (dsherret) [inline cli/args/mod.rs:823]: I think the code block below and the one in the language server are essentially the same thing? Could we maintain this snapshot_from_lockfile function in the CLI and then just have that call the deno_npm functions?

**Review by dsherret — COMMENTED** [review]
Almost there.

File: cli/args/mod.rs (hunk 2 above)
Comment (magurotuna) [reply cli/args/mod.rs:822]: Sounds good, how about this [9aa1e2d](https://github.com/denoland/deno/pull/20024/commits/9aa1e2ddf9bd378e9a058085fc294b64875bb35b)

**Pull request merged**